File: hello-world/src/main/java/kz/sabyr/quarkus/helloworl/MyResource.java
package kz.sabyr.quarkus.helloworl;

import kz.sabyr.quarkus.service.MyService;

import javax.inject.Inject;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import java.util.Map;

@Path("/hello-world")
public class MyResource {
    @Inject
    MyService myService;

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return "hello";
    }

    @GET
    @Path("/{name}")
    @Produces(MediaType.APPLICATION_JSON)
    public Map<String, String> helloWithParam(@PathParam(value = "name") String name) {
        return Map.of("name", name + ", ID: " + myService.generateUUID());
    }
}

File: hello-world-kotlin/src/main/kotlin/kz/sabyr/quarkus/helloworlkotlin/MyResource.kt
package kz.sabyr.quarkus.helloworlkotlin

import kz.sabyr.quarkus.helloworlkotlin.model.MyData
import kz.sabyr.quarkus.helloworlkotlin.service.MyService
import javax.enterprise.inject.Default
import javax.inject.Inject
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.PathParam
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType

@Path("/hello-world")
class MyResource {
    @Inject
    @field: Default
    lateinit var myService: MyService

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    fun hello() = "hello"

    @GET
    @Path("/{name}")
    @Produces(MediaType.APPLICATION_JSON)
    fun hello(@PathParam("name") name: String) = MyData(name + ", ID: " + myService.generateUUID())
}